lib/extensions/ar_region.rb

```diff
@@ -117,6 +117,13 @@ def partition_objs_by_remote_region(objs)
     def partition_ids_by_remote_region(ids)
       ids.partition { |id| self.id_in_current_region?(id) }
     end
+
+    def discover_my_region_number
+      ENV.fetch("REGION", nil).try(&:to_i) ||
+        File.read(File.join(Rails.root, "REGION")).to_i rescue nil ||
+        # note: nil.to_i == 0
+        connection.select_value("select id from miq_databases limit 1").to_i / DEFAULT_RAILS_SEQUENCE_FACTOR
```

Comment:
Also make this a private method if it's not already (hard to tell from the diff)